pkg/apis/enterprise/v2/common_types.go

```diff
@@ -288,6 +314,7 @@ type AppDeploymentInfo struct {
 	Size             uint64              `json:"Size,omitempty"`
 	RepoState        AppRepoState        `json:"repoState"`
 	DeployStatus     AppDeploymentStatus `json:"deployStatus"`
+	AppInstallStatus AppInstallStatus    `json:"appInstallStatus"`
```

Comment:
I know, this is the first code we are getting in, but, going forward, I think we need to break up the `AppInstallStatus` into multiple phases(probably slightly different naming, say `podInstall`, `podCopy`, `operatorDownload`, etc..), and each phase pointing to separate auxiliary status info.  This way, it will be clear to move the phases on error conditions. For example, if the phase is `podInstall`, and if the app package is not found on the Pod, can be moved to phase `podCopy`. If the app package is not found in the phase `podCopy`, the phase can be moved to 'operatorDownload` etc..

Also,  it will be much easy to decouple the workers, and also to accommodate for any future functionality needs, if we want to introduce in between.

Reply:
Yeah that's the plan going forward, and that is the reason we created a struct for the states. Sorry for the terminology but in `AppInstallStatus ` we can have something like `AppDownloadstate`, `AppCopyState` and `AppInstallState` for things to be more clear.